js/server/modules/@arangodb/testutils/aql-graph-traversal-generic-tests.js

```diff
@@ -1294,6 +1294,43 @@ function testOpenDiamondDfsLabelVariableForwarding(testGraph) {
   }
 }
 
+function testOpenDiamondPathVarOptimization(testGraph, mode) {
+  assertTrue(testGraph.name().startsWith(protoGraphs.openDiamond.name()));
+
+  const buildQuery = (returnFormat) => {
+    return `
+      FOR v,e,p IN 1..3 ANY "${testGraph.vertex('A')}"
+      GRAPH "${testGraph.name()}" OPTIONS {order: "${mode}"}
+      RETURN ${returnFormat}
+    `;
+  };
+
+  // With that list of return 'formats' we do expect that the optimizer-rule:
+  // "remove-redundant-path-var" is being executed.
+  const returnFormatsRuleShouldAppear = [
+    '[v]', '[e]', '[v, e]', '[e, v]'
+  ];
+  const optimizerRuleToDeactivate = ["-optimize-traversals"];
+
+  let queryList = [];
+  for (const returnFormat of returnFormatsRuleShouldAppear) {
+    queryList.push(buildQuery(returnFormat));
+  }
+
+  for (const query of queryList) {
+    const optPlans = AQL_EXPLAIN(query, {}, { optimizer: { rules: optimizerRuleToDeactivate } }).plan;
+    assertTrue(optPlans.rules.includes("remove-redundant-path-var"));
+  }
+
+  // Now return p, to make sure the rule is not being executed.
+  const optPlans = AQL_EXPLAIN(buildQuery('[p]'), {}, { optimizer: { rules: optimizerRuleToDeactivate } }).plan;
+  assertEqual(optPlans.rules, []);
```

Comment:
nit: this makes the test depend on if there are other, unrelated optimizer rules than the one we are looking for.
This may break the test when new optimizer rules are added.
Would prefer a more tailored solution that checks if the optimizer rule in question is not there, but will be happen with other, unrelated optimizer rules.
```suggestion
  assertFalse(optPlans.rules.includes("remove-redundant-path-var"));
```

Reply:
Yes, my implementation is very "strict". Let me think about this. I think I might like the approach that a new optimizer rule may break my test - to be eventually able to identify an unwanted change. Changing it to your suggestion works, but would not identify any potentially unwanted optimizer rule changes for that simply query...

Reply:
Ok, CI proved that your approach is required. Will change.